pages/Javascript-API.md

```diff
@@ -56,13 +89,69 @@ The properties of the feature at the current cursor location may be accessed thr
 
-#### `requestRedraw()`
-Requests an update to the drawn map. If `animated: true` is set, this happens once per frame automatically.
-
 #### `getActiveCamera()`
+Returns the active camera.
 
-#### `setActiveCamera()`
+#### `requestRedraw()`
+Requests an update to the drawn map. If `animated: true` is set, this happens once per frame automatically.
 
 #### `reload(scene_url)`
+Loads the specified scene by url and rebuilds the geometry.
 
 #### `rebuild()`
+Rebuilds the current scene from scratch.
+
+#### `screenshot()`
+This queues a screenshot request, returning a Promise that fulfills when the screenshot is available. (The screenshot must be "queued" because it cannot be captured immediately: we must ensure that the GL buffer is finished drawing, and then must capture the buffer contents just after rendering, before it is cleared by other operations.)
+
+The promise resolves with an object containing two properties:
+
+- url: a data URL of the Canvas contents, suitable for loading into an <img> or opening in a new tab/window
+- blob: a Blob of type image/png, suitable for saving to a file, either manually or with a third-party library such as FileSaver.js
+
+```javascript
+scene.screenshot().then(function(screenshot) {
+    // uses FileSaver.js: https://github.com/eligrey/FileSaver.js/
+    saveAs(screenshot.blob, 'tangram-' + (+new Date()) + '.png');
+});
+```
+
+#### `setActiveCamera(_string_ camera)`
+Sets the active camera to the camera specified by name, as named in the scenefile.
+
+#### `setDataSource(_string_ name, _object_ config)`
+Loads a new `source` object (see `[sources](sources.md)`).
+
+If "name" doesn't match an existing source, a new source object will be created. The "config" object must follow the `[sources](sources.md#sources)` specification.
+
+```javascript
+scene.setDataSource("osm", {type: 'TopoJSON', url: "//vector.mapzen.com/osm/all/{z}/{x}/{y}.topojson" });
+```
+
+```javascript
+var geojson_data = {};
+...
+scene.setDataSource("dynamic_data", {type: 'GeoJSON', data: geojson_data });
+```
+
+#### `updateConfig()`
+Re-parses the scene.config object and rebuilds the scene from scratch, updating data sources, reloading textures, and rebuilding geometry.
+
+```javascript
+scene.updateConfig()
+```
+
+#### `scene.view_complete`
+This is an event which fires when the view is in a "resting state", meaning new geometry is rendered, and no further tiles are loading/building. For example, when a scene is loaded, a view_complete event will fire when all tiles have loaded and the map renders. If the view is then zoomed in a level, another view_complete event will fire when the next zoom finishes rendering.
+
+`view_complete` can be subscribed to like other scene events:
+
+```javascript
+scene.subscribe({
+    view_complete: function () {
+        console.log('scene view complete, rendered ' +
+            scene.render_count + ' primitives at: ' +
```

Comment:
rather than officially exposing properties like `scene.render_count`, maybe just make this a simple console messages without internal properties.